### PR TITLE
Add support for device-specific weston config (#499)

### LINKED ISSUE #499: Weston config should be loaded on start
Currently the config of weston is loaded post-install. This is done in [postmarketos-ui-weston.post-install](https://github.com/postmarketOS/pmbootstrap/blob/314c17e03cf8cddfd0f385d9db2f23f76f9a0418/aports/main/postmarketos-ui-weston/postmarketos-ui-weston.post-install).

Whenever changing one of the weston config values in the deviceinfo files, you should explicitly reinstall `postmarketos-ui-weston` (this is not done automatically). However, this does not make much sense because the deviceinfo file is in a `device-*` package, so you suspect only having to rebuild the `device-*` package. 

Therefore I propose to load the weston config when starting weston, in the [startup_weston.sh script](https://github.com/postmarketOS/pmbootstrap/blob/314c17e03cf8cddfd0f385d9db2f23f76f9a0418/aports/main/postmarketos-ui-weston/start_weston.sh).

### PR DESCRIPTION
**Pull request by craftyguy**
Fixes #499. See this issue for more details

### CODE REVIEW

**Comment by craftyguy** [comment]
This currently doesn't work due to #731

**Comment by montvid** [comment]
nano /tmp/weston.log

> Old Xwayland module loading detected: Please use --xwayland command line option or set xwayland=true in the [core] section in weston.ini

Please change from modules=xwayland.so to xwayland=true It would be nice if it was changed in the default weston.ini too.

**Comment by craftyguy** [comment]
There is no longer going to be a 'default weston.ini' file, that's the whole point of this PR and the discussion in #499.

**Comment by montvid** [comment]
Well I don't think it is practical to have a weston.ini for every device as not many people will be using weston anyway as they need a more proper gui.

**Comment by craftyguy** [comment]
Go read the discussion in #499, and add any additional comments you might have in that issue.

**Comment by drebrez** [comment]
@craftyguy closing this PR will break weston in all the devices except N900? 
Or weston may work correctly without the config?

File: aports/device/device-nokia-rx51/APKBUILD [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+pkgrel=24
 pkgdesc="Nokia N900"
 url="https://github.com/postmarketOS"
 arch="noarch"
 license="MIT"
 depends="linux-postmarketos uboot-tools linux-firmware kbd kbd-bkeymaps ofono"
 makedepends="uboot-tools kbd kbd-bkeymaps"
 install="$pkgname.post-install"
-subpackages=""
+subpackages="$pkgname-weston:weston:noarch"
```
Comment (ollieparanoid) [inline aports/device/device-nokia-rx51/APKBUILD]: The `:noarch` is unnecessary when the parent package has the same architecture.

**Comment by craftyguy** [comment]
Oh shit, what did I volunteer for? 😛 

I would appreciate some help adding this to device packages.. since there are a lot.

**Comment by ollieparanoid** [comment]
Here we go! I have no idea why one Travis check is failing though, I didn't even touch `firmware-huawei-y530` in this PR, and also the checksum are correct in master.

Reviewing this PR and testing it on real devices would be greatly appreciated. I've quickly tested it in `qemu-amd64` and it worked for me.

**Comment by craftyguy** [comment]
wow, nice work @ollieparanoid !

File: aports/main/postmarketos-ui-weston/weston.ini.default [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# In postmarketOS, this file gets used, when no /etc/xdg/weston.ini exists. That 
+# is, so we can install a device specific config file (which may disable 
+# xwayland for example) from the device specific package to /etc/xdg/weston.ini.
+
+[core]
+modules=xwayland.so
```
Comment (drebrez) [inline aports/main/postmarketos-ui-weston/weston.ini.default]:
```
[core]
xwayland=true
```
is not necessary here?

File: aports/device/device-oneplus-bacon/weston.ini [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# XWayland seems to be broken for this device.
```
Comment (drebrez) [inline aports/device/device-oneplus-bacon/weston.ini]: missing changes to the `aports/device/device-oneplus-bacon/APKBUILD`?

File: aports/main/postmarketos-ui-weston/weston.ini.default (hunk 2 above)
Comment (ollieparanoid) [reply aports/main/postmarketos-ui-weston/weston.ini.default]: Thanks! `modules=xwayland.so` is the obsolete version, `xwayland=true` is the proper one, see https://github.com/postmarketOS/pmbootstrap/pull/739#issuecomment-336144375 -- I'll adjust this.

File: aports/device/device-oneplus-bacon/weston.ini (hunk 3 above)
Comment (ollieparanoid) [reply aports/device/device-oneplus-bacon/weston.ini]: Good catch, fixed.

**Comment by ollieparanoid** [comment]
@drebrez wrote:
> `/etc/xdg/weston.ini` is not the default location that weston looks, if no `-c ...weston.ini` parameter is specified, it uses `/etc/xdg/weston/weston.ini` 

- [ ] use `/etc/xdg/weston/` for the configs

**Comment by drebrez** [comment]
After a lot of testing and git bisecting, I decided to rebuild all the packages from scratch and finally only by adding `backend=fbdev-backend.so` to the `weston.ini.default` everything started working again

**Comment by ollieparanoid** [comment]
Wow, that you so much for figuring that out! I'll adjust the PR accordingly.

**Comment by drebrez** [comment]
@ollieparanoid i'm doing it right now

**Comment by ollieparanoid** [comment]
Even better, thanks \o/

**Pull request merged**